### DISCUSSION TITLE
topological sort

### DISCUSSION THREAD

**Discussion by majidaldo** [tweag/nickel#1348] [Q&A]
given a dependency graph, how does one go about doing a topological sort?

**Comment by yannham** [comment]
Depends on how your dependency graph is represented, but some time ago we translated (in a somehow mechanic way, although by hand) the Nixpkgs list library to Nickel for benchmarking, and the library does implement a topological sort: https://github.com/tweag/nickel/blob/ce9501df692ca872792981b888a1940bf5a44b61/benches/nixpkgs/lists.ncl#L433

Note that this code is used to benchmark the typechecker, so it's not actually tested for correctness in practice. But this can be a good starting point!
